lib/active_merchant/billing/gateways/paypal_express.rb

```diff
@@ -147,6 +147,8 @@ def build_setup_request(action, money, options)
               if options[:allow_guest_checkout]
                 xml.tag! 'n2:SolutionType', 'Sole'
                 xml.tag! 'n2:LandingPage', options[:landing_page] || 'Billing'
+              elsif options[:paypal_chooses_landing_page]
```

Comment:
Would make sense to leverage the `allow_guest_checkout` for this?

Reply:
sure, maybe I can pass `paypal_chooses_landing_page` too and that would result in

`xml.tag! 'n2:LandingPage', options[:landing_page] || 'Billing' unless options[:paypal_chooses_landing_page` ?